apps/studio/components/interfaces/Settings/General/General.tsx

```diff
@@ -42,27 +57,36 @@ const General = () => {
       },
     }
   )
+  const isBranch = parentProject !== undefined
+  const form = useForm<z.infer<typeof schema>>({
+    resolver: zodResolver(schema),
+    defaultValues: { name: '' },
+    // when `useSelectedProjectQuery finishes loading, it will update
+    // this value
+    values: { name: project?.name ?? '' },
+  })
 
-  const { mutate: updateProject, isLoading: isUpdating } = useProjectUpdateMutation()
+  const { mutate: updateProject } = useProjectUpdateMutation()
 
-  const onSubmit = async (values: any, { resetForm }: any) => {
+  const onSubmit: SubmitHandler<z.infer<typeof schema>> = async ({ name }) => {
     if (!project?.ref) return console.error('Ref is required')
 
     updateProject(
-      { ref: project.ref, name: values.name.trim() },
+      { ref: project.ref, name },
       {
         onSuccess: ({ name }) => {
-          resetForm({ values: { name }, initialValues: { name } })
+          form.reset({ name }, { keepValues: true })
           toast.success('Successfully saved settings')
         },
       }
     )
   }
 
   return (
-    <div>
+    <ScaffoldSection className="gap-6">
+      <ScaffoldSectionTitle>General settings</ScaffoldSectionTitle>
```

Comment:
changed this to a section title for consistency with the rest of the page, where every other section has a title